Automattic/wp-calypso
# Add logo upload section to website content page

#### Changes proposed in this Pull Request

* Adds logo upload step section in the accordion
* Adds state and related code (actions, reducers) for logo url state
* Refactors the state so that website content includes pages as well as a logo url
* Adds unit tests for logo url state
* Adds unit tests for handling logo upload state
* Adds other image related missing unit tests
* A minor bug fix where the website-content step was visible when redirecting on website-content submitted.

![logoupload](https://user-images.githubusercontent.com/3422709/155655249-69cd9670-f474-4d37-b663-06e2f277a269.gif)


#### Testing instructions

* Make sure all unit tests are working

* Test Logo upload Functionality
    * Go through DIFM flow, checkout and come to the website-content filling step
    * Upload a logo
    * Make sure the thumbnail is visible
    * Upload other image and make sure they are visible
    * Submit form and make sure lgog url and other details are properly shown in the site build post
 
* Upload a non image file and make sure the upload failure is visible and that everything else continues to work
* On a website content submitted site try to access the `/start/website-content` page.
     * The website content should not be briefly visible before redirecting.

Related to #
